pkg/client/cli/intercept_state.go

```diff
@@ -110,6 +112,22 @@ func leaveCommand() *cobra.Command {
 	}
 }
 
+// isAirGapped let's us know if the user is able to resolve a connection
+// to SystemA.
+func isAirGapped(ctx context.Context) (isAirGapped bool) {
+	env, err := client.LoadEnv(ctx)
+	if err == nil {
+		timeout := 2 * time.Second
+		conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%s", env.SystemAHost, env.SystemAPort), timeout)
+		if err != nil {
+			isAirGapped = true
+			return
+		}
+		conn.Close()
+	}
+	return
+}
```

Comment:
Everyone keeps saying "air-gapped", but I think the scenario is actually just the cluster having a locked-down firewall.  I think the laptop is still able to talk to the Internet; so I'm not sure this is the right place for this check.